golem-worker-executor-base/src/worker.rs

```diff
@@ -86,6 +89,41 @@ pub struct Worker<Ctx: WorkerCtx> {
     oom_retry_config: RetryConfig,
 }
 
+impl<Ctx: WorkerCtx> Worker<Ctx> {
+    pub async fn get_component_metadata(
+        worker: &Arc<Worker<Ctx>>,
+    ) -> Result<(WorkerMetadata, Component, ComponentMetadata), GolemError> {
+        let component_id = worker.owned_worker_id.component_id();
+        let worker_metadata = worker.get_metadata().await?;
+
+        let component_version = worker_metadata
```

Comment:
This decision of whether we are attempting an update or not should not be part of the `get_component_metadata`, it is something we have to decide, and log, once when the component starts